orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -2220,27 +2220,26 @@ Return true if all elements are true and false otherwise.
       .Output(0, "outputs_grad", "Gradient of outputs returned from pytorch.", "T", OpSchema::Variadic,
               /*is_homogeneous*/ false,
               /*min_arity*/ 1)
-      .Attr(
-        "required_grad",
-        "The indices of the outputs that require gradient outputs.",
-        AttributeProto::INTS)
+      .Attr("full_shape_outputs", "The indices of the outputs that must have full shape.", AttributeProto::INTS)
```

Comment:
also add an assert in yield's kernel?
output shape should match input shape if "full_shape" is true.